internal/app/components/courses/courseRecords.go

```diff
@@ -68,6 +68,19 @@ func (cr CourseRecords) Exclude(records CourseRecords) CourseRecords {
 	return result
 }
 
+// calculates the cumulative average of completed courses
+func (cr CourseRecords) CurrentCAV() CourseGrade {
+	total, count := CourseGrade(0), 0
+	for _, record := range cr {
+		// only accumulate if course is completed
+		if record.IsCompleted() {
+			total += record.Grade
+			count += 1
+		}
+	}
+	return CourseGrade(int(total) / count)
```

Comment:
Should we consider the case where `count` is 0 because no courses are completed?

Reply:
Yeah we do, go catch